### PR TITLE
Update the CEL Grammar to support the lexis for Google SQL strings

### PR DESCRIPTION
**Pull request by TristonianJones**
#45 The grammar has been updated to only support escape sequences
supported within [Google SQL](https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical).

The set of numeric literals supported by the grammar is a superset of those
supported within Google SQL (namely the ability to specify a hex exponent),
but whether the grammar should exclude such numeric values is unclear since
this is simply a syntactic matter and does not relate to any specific encoding.

### CODE REVIEW

File: parser/gen/CEL.g4 [hunk 1]
```diff
@@ -173,8 +170,8 @@ NUM_UINT
    ;
 
 STRING
-  : '"' (ESC_SEQ | ~('"'|'\n'|'\r'))* '"'
-  | '\'' (ESC_SEQ | ~('\''|'\n'|'\r'))* '\''
+  : '"' (ESC_SEQ | ~('\\'|'"'|'\n'|'\r'))* '"'
```
Comment (JimLarson) [inline parser/gen/CEL.g4:173]: Use "BACKSLASH" for readability, here and below?
Consider defining DQUOTE and SQUOTE too.

File: parser/gen/CEL.g4 [hunk 2]
```diff
@@ -183,8 +180,6 @@ STRING
   | RAW '\'\'\'' .*? '\'\'\''
   ;
 
-fragment RAW : 'r' | 'R';
-
 BYTES : ('b' | 'B') STRING;
 
 IDENTIFIER : (LETTER | '_') ( LETTER | DIGIT | '_')*;
```
Comment (JimLarson) [inline parser/gen/CEL.g4]: No newline at end of file.  (Fix your IDE! :-)

File: parser/unescape_test.go [hunk 3]
```diff
@@ -112,3 +112,20 @@ func TestUnescapeUnicodeSequence(t *testing.T) {
...
+	if err != nil {
+		t.Error(err)
+	}
+	if text != "\a\b\f\n\r\t\v'\"\\? Legal escapes" {
+		t.Errorf("Got '%v', wanted '%v'", text, "\a\b\f\n\r\t\v'\"\\? Legal escapes")
+	}
+}
+
+func TestUnescapeIllegalEscapes(t *testing.T) {
+	text, err := unescape(`"\a\b\f\n\r\t\v\'\"\\\? Illegal escapes \>"`)
```
Comment (JimLarson) [inline parser/unescape_test.go]: Make it clear that the only illegal escape is backslash-greaterthan.

File: parser/unescape_test.go [hunk 4]
```diff
@@ -112,3 +112,20 @@ func TestUnescapeUnicodeSequence(t *testing.T) {
...
+	if err == nil {
+		t.Errorf("Got '%v', expected error", text)
+	}
+}
```
Comment (JimLarson) [inline parser/unescape_test.go]: No newline at end of file.

**Review by JimLarson — CHANGES_REQUESTED** [review]
Update cel-spec/doc/langdef.md if necessary.

File: parser/gen/CEL.g4 (hunk 1 above)
Comment (TristonianJones) [reply parser/gen/CEL.g4:173]: I can't unfortunately, I tried. Lexer fragments don't work within the alternative groups.

File: parser/gen/CEL.g4 (hunk 2 above)
Comment (TristonianJones) [reply parser/gen/CEL.g4]: Done. :)

**Comment by TristonianJones** [comment]
Thanks Jim. I'll update the `langdoc.md` once this change is submitted.

**Review by JimLarson — APPROVED** [review]

**Pull request merged**